pkg/controller/snapshot_controller.go

```diff
@@ -659,37 +667,30 @@ func (ctrl *csiSnapshotController) createSnapshotOperation(snapshot *crdv1.Volum
 		return nil, err
 	}
 	// Create VolumeSnapshotContent in the database
-	volumeRef, err := ref.GetReference(scheme.Scheme, volume)
-	if err != nil {
-		return nil, err
-	}
 	snapshotRef, err := ref.GetReference(scheme.Scheme, snapshot)
 	if err != nil {
 		return nil, err
 	}
 
-	if class.DeletionPolicy == nil {
-		class.DeletionPolicy = new(crdv1.DeletionPolicy)
-		*class.DeletionPolicy = crdv1.VolumeSnapshotContentDelete
-	}
 	timestamp := creationTime.UnixNano()
 	snapshotContent := &crdv1.VolumeSnapshotContent{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: contentName,
 		},
 		Spec: crdv1.VolumeSnapshotContentSpec{
-			VolumeSnapshotRef:   snapshotRef,
-			PersistentVolumeRef: volumeRef,
-			VolumeSnapshotSource: crdv1.VolumeSnapshotSource{
-				CSI: &crdv1.CSIVolumeSnapshotSource{
-					Driver:         driverName,
-					SnapshotHandle: snapshotID,
-					CreationTime:   &timestamp,
-					RestoreSize:    &size,
-				},
+			VolumeSnapshotRef: *snapshotRef,
+			DeletionPolicy:    class.DeletionPolicy,
+			Driver:            driverName,
+			SnapshotClassName: &class.Name,
+			Source: crdv1.VolumeSnapshotContentSource{
+				VolumeHandle: &volume.Spec.CSI.VolumeHandle,
 			},
-			VolumeSnapshotClassName: &(class.Name),
-			DeletionPolicy:          class.DeletionPolicy,
+		},
+		Status: crdv1.VolumeSnapshotContentStatus{
```

Comment:
I think Status can't be created together with the content this way.  Status can only be updated later as subresource.

Reply:
updated. thx